### PR TITLE
Closes #1133 Adding required markup for subhead and moving to the top of flexible page migration.

### LINKED ISSUE #1133: Fix page subtitle migration
## Motivation
When running a migration, the subtitle field is appearing as the last paragraph instead of the first. The subtitle is also not styled as expected.

## Context
Original goal was to migrate from the specific field to an az_paragraphs_text paragraph with surrounding styles `<p class="text-lead">{content}</p>`

[az_paragraph_flexible_page_sub](https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_flexible_page_sub.yml) file controls this migration

## Examples
- [UAQS Flexible Page](https://31da50fb-682c-4cd5-a9e8-f157c8b821a9--site-migration.probo.build/node/34) subtitle out of order
- [Primus page](https://31da50fb-682c-4cd5-a9e8-f157c8b821a9--site-migration.probo.build/node/27) subtitle is out of order

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Adding required markup via a new process plugin.
Added subhead before the rest of the paragraphs in the page migration.

## Related Issue
Closes #1133 

## How Has This Been Tested?
Locally
Probo
image

You can see it working here: https://77050b59-93ad-45f2-b313-d805725916d7--site-migration.probo.build/uaqs_flexible_page

## Types of changes

**Arizona Quickstart** (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

**Drupal core**
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

**Drupal contrib projects**
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/src/Plugin/migrate/process/ConvertStringToLeadPTag.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *
+ * @code
+ *   process:
+ *     field_example:
+ *       plugin: convert_string_to_lead_p_tag
+ *       source: example
+ * @endcode
+ *
+ * @MigrateProcessPlugin(
+ *   id = "convert_string_to_lead_p_tag"
```
Comment (joeparsons) [inline modules/custom/az_migration/src/Plugin/migrate/process/ConvertStringToLeadPTag.php:24]: Nitpick, we should prefix the machine name with `az_`.

Comment (joeparsons) [inline modules/custom/az_migration/src/Plugin/migrate/process/ConvertStringToLeadPTag.php:24]:
```suggestion
 *   id = "az_convert_string_to_lead_p_tag"
```

**Comment by joeparsons** [comment]
General question: is a process plugin really needed?  Maybe we could just use `concat`?

**Comment by trackleft** [comment]
OK, removed the custom plugin

**Review by joeparsons — DISMISSED** [review]
Looks ready to merge now IMHO.

**Comment by trackleft** [comment]
FYI I changed this from using the `sub_process` plugin to `field_name/0/value` because I couldn't get the `sub_process` plugin to find the constants for some reason.

**Comment by trackleft** [comment]
Also tried using the `dom_apply_style` plugin, but that didn't work with plain text from the source.

It did work when i prewrapped the source with a `p` tag before importing into PHP DOM, but then it converted it to a span and added the correct class, but that doesn't match the requirements, so I stuck with concat.

**Comment by joeparsons** [comment]
> FYI I changed this from using the `sub_process` plugin to `field_name/0/value` because I couldn't get the `sub_process` plugin to find the constants for some reason.

I'm guessing this might have worked if the `include_source` config option for the `sub_process` plugin were enabled.

**Comment by trackleft** [comment]
> I'm guessing this might have worked if the `include_source` config option for the `sub_process` plugin were enabled.

I'm testing this

**Comment by trackleft** [comment]
> > I'm guessing this might have worked if the `include_source` config option for the `sub_process` plugin were enabled.

This worked!

**Review by joeparsons — APPROVED** [review]

**Review by jwmoore1 — APPROVED** [review]

**Review by akslay — APPROVED** [review]

**Pull request merged**